build/build.go

```diff
@@ -205,15 +205,6 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opts map[
 		return nil, err
 	}
 
-	defers := make([]func(), 0, 2)
-	defer func() {
-		if err != nil {
-			for _, f := range defers {
-				f()
-			}
-		}
-	}()
```

Comment:
Curious if this approach (collect the `release` functions and a single `defer` would still be preferred, but possibly without the `if err != nil` ? I know some linters will complain about the `defer release()` in a loop.